src/cookieSyncManager.ts

```diff
@@ -159,4 +202,30 @@ export const isLastSyncDateExpired = (
         new Date().getTime() >
         new Date(lastSyncDate).getTime() + frequencyCap * DAYS_IN_MILLISECONDS
     );
-};
\ No newline at end of file
+};
+
+export const isTcfApiAvailable = (): boolean => isFunction(window.__tcfapi);
+
+export async function appendGdprConsentUrl(url: string, logger: Logger): Promise<string> {
+    return new Promise((resolve, reject) => {
+        try {
+            const tcfAPICallBack = (inAppTCData: TCData, success: boolean) => {
+                if (success && inAppTCData) {
+                    const gdprApplies = inAppTCData.gdprApplies ? 1 : 0;
+                    const tcString = inAppTCData.tcString;
+                    resolve(`${url}&gdpr=${gdprApplies}&gdpr_consent=${tcString}`);
+                } else {
+                    resolve(url); // No GDPR data; fallback to original URL
+                }
+            }
+
+            // `getInAppTCData` is the function name
+            // 2 is the version of TCF (2.2 as of 1/22/2025)
+            // callback 
+            window.__tcfapi('getInAppTCData', 2, tcfAPICallBack);
```

Comment:
Might be a nitpick on my part but I think this makes it more readable and obvious.

```suggestion
            window.__tcfapi(
                'getInAppTCData',  // the function name
                2,                              // Version of TCF (2.2 as of 1/22/2025)
                tcfAPICallBack,       // callback function
            );
```